src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,267 @@
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  std::string column_delimiter;
+  if (CLI::HasParam("column_delimiter"))
+  {
+    column_delimiter = CLI::GetParam<std::string>("column_delimiter");
+    // Allow only 3 delimiters.
+    RequireParamValue<std::string>("column_delimiter", [](std::string del)
+        { return del == "\t" || del == "," || del == " "; }, true,
+        "Delimiter should be either \\t (tab) or , (comma) or ' ' (space) ");
+  }
+  else
+  {
+      if (data::Extension(filename) == "csv")
+      {
+        column_delimiter = ",";
+        Log::Warn << "Found csv Extension, taking , as column_delimiter. \n";
+      }
+      else if (data::Extension(filename) == "tsv" ||
+          data::Extension(filename) == "txt")
+      {
+        column_delimiter = "\t";
+        Log::Warn << "Found tsv or txt Extension, taking \\t as"
+        "column_delimiter. \n";
+      }
+      else
+      {
+        Log::Warn << "column_delimiter not specified, taking default value \n";
+      }
+  }
+  // Handling Dimension vector
+  std::vector<std::string> temp_dimension =
+      CLI::GetParam<std::vector<std::string> >("dimension");
+  std::set<size_t>dimension;
+  int columnstartindex, columnendindex;
+  size_t found;
+  for (size_t i = 0; i < temp_dimension.size(); i++)
+  {
+    found = temp_dimension[i].find('-');
+    if ( found != string::npos)
+    {
+      try
+      {
+        // Has a range include, something of type a-b.
+        columnstartindex = std::stoi(temp_dimension[i].substr(0, found));
+        columnendindex = std::stoi(temp_dimension[i].substr(found+1,
+            temp_dimension[i].length()));
+      }
+      catch (const std::exception& e)
+      {
+        Log::Fatal << "Dimension value not clear, either negatve or can't "
+        "parse the range. Usage a-b \n";
+      }
+      for (int i = columnstartindex; i <= columnendindex; i++)
+        dimension.insert(i);
+    }
+    else
+    {
+      try
+      {
+        dimension.insert(std::stoi(temp_dimension[i]));
+      }
+      catch (const std::exception& e)
+      {
+        Log::Fatal << "Dimension value not appropriate \n";      
+      }
+    }
+  }
+
+  // Extracting the Contents of file
+  // File pointer
+  ifstream fin;
+  // Open an existing file
+  fin.open(filename);
+  if (!fin.is_open())
+  {
+    Log::Fatal << "Unable to open input file \n";
+  }
+  std::string line, word;
+  std::vector<std::vector<std::string>> dataset;
+  boost::string_view token, copy;
+  while (std::getline(fin, line))
```

Comment:
This loop can be moved to a separate function that accepts `filename` and returns `dataset`.